lib/ebay_api/middlewares/save_to_file_response.rb

```diff
@@ -0,0 +1,29 @@
+require "securerandom"
+
+# Parses gzipped response
+class EbayAPI
+  class SaveToFileResponse
+    def initialize(app)
+      @app = app
+    end
+
+    def call(env)
+      status, headers, body = @app.call(env)
+
+      [status, headers, body.map { |b| save(b) }]
+    end
+
+    private
+
+    def save(body)
+      return unless body
+      return if body.empty?
+
+      file = File.new("/tmp/ebay-api-response-#{SecureRandom.hex}", "w+b")
+      file.write(body.to_s)
```

Comment:
What is `body` here? If it is some stream then calling `#to_s` will force it to “materialize” in memory (and there can be not enough memory)

Can it be removed?
```suggestion
      file.write(body)
```